framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1029,6 +1029,42 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
         // check precondition annotations
         checkPreconditions(node, contractsUtils.getPreconditions(invokedMethodElement));
 
+        // Checks that the following rule is satisfied:
+        // The type on a constructor declaration must be a supertype of
+        // the return type of "super()" invocation within that constructor.
+        if (TreeUtils.isSuperCall(node)) {
+            TreePath path = atypeFactory.getPath(node);
+            MethodTree enclosingMethod = TreeUtils.enclosingMethod(path);
+            if (TreeUtils.isConstructor(enclosingMethod)) {
+                AnnotatedTypeMirror superType = atypeFactory.getAnnotatedType(node);
+                AnnotatedExecutableType constructorType =
+                        atypeFactory.getAnnotatedType(enclosingMethod);
+                Set<? extends AnnotationMirror> topAnnoations =
+                        atypeFactory.getQualifierHierarchy().getTopAnnotations();
+                for (AnnotationMirror topAnno : topAnnoations) {
+                    AnnotationMirror superTypeMirror = superType.getAnnotationInHierarchy(topAnno);
+                    AnnotationMirror constructorTypeMirror =
+                            constructorType.getReturnType().getAnnotationInHierarchy(topAnno);
+                    // Checking AnnotaionUtils.AreSameByName for the following case:
+                    // Example from tests/nullness/DaikonTests.java: "class Bug2 extends Bug2Super"
+                    // Here, constructorTypeMirror is
+                    // @UnderInitialization(DaikonTests.Bug2Super.class)
```

Comment:
I do not understand from this why this uses `areSameByName`. Instead of the giving the specific test case, could you try to explain the general logic?

Couldn't there be an example where the `@UnderInitialization` annotation is used incorrectly and an error should be raised, even though they have the same annotation name?